plat/common/bootinfo.c

```diff
@@ -113,3 +119,142 @@ void ukplat_bootinfo_print(void)
 			   );
 	}
 }
+
+#ifdef CONFIG_LIBFDT
+static void fdt_bootinfo_mem_mrd(struct ukplat_bootinfo *bi, void *fdtp)
+{
+	struct ukplat_memregion_desc mrd = {0};
+	int prop_len, prop_min_len;
+	__u64 mem_base, mem_sz;
+	int naddr, nsz, nmem;
+	const __u64 *regs;
+	int rc;
+
+	/* search for assigned VM memory in DTB */
+	rc = fdt_check_header(fdtp);
+	if (unlikely(rc))
+		ukplat_bootinfo_crash("Reserved memory is not supported");
+
+	nmem = fdt_node_offset_by_prop_value(fdtp, -1, "device_type",
+					     "memory", sizeof("memory"));
+	if (unlikely(nmem < 0))
+		ukplat_bootinfo_crash("No memory found in DTB");
+
+	naddr = fdt_address_cells(fdtp, nmem);
+	if (unlikely(naddr < 0 || naddr >= FDT_MAX_NCELLS))
+		ukplat_bootinfo_crash("Could not find proper address cells!");
+
+	nsz = fdt_size_cells(fdtp, nmem);
+	if (unlikely(nsz < 0 || nsz >= FDT_MAX_NCELLS))
+		ukplat_bootinfo_crash("Could not find proper size cells!");
+
+	/*
+	 * QEMU will always provide us at least one bank of memory.
+	 * unikraft will use the first bank for the time-being.
+	 * The property must contain at least the start address
+	 * and size, each of which is 8-bytes.
+	 * TODO: Support more than one memory@ node/regs/ranges properties.
+	 */
+	prop_len = 0;
+	prop_min_len = (int)sizeof(fdt32_t) * (naddr + nsz);
+	regs = fdt_getprop(fdtp, nmem, "reg", &prop_len);
+	if (unlikely(!regs || prop_len < prop_min_len))
+		ukplat_bootinfo_crash("Bad 'reg' property or more than one memory bank.");
+
+	mem_sz = fdt64_to_cpu(regs[1]);
+	mem_base = fdt64_to_cpu(regs[0]);
+	if (unlikely(!RANGE_CONTAIN(mem_base, mem_sz,
+				    __BASE_ADDR, __END - __BASE_ADDR)))
+		ukplat_bootinfo_crash("Image outside of RAM");
+
+	mrd.vbase = (__vaddr_t)mem_base;
+	mrd.pbase = (__paddr_t)mem_base;
+	mrd.len   = __BASE_ADDR - mem_base;
```

Comment:
Very good point! I thought it would not matter since `ukplat_paging_init` ignores such regions and nothing would break. But yeah, the memory map ends up having an unnecessary memory region descriptor. Will add a check for it!